plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEngineServerLaunchConfigurationDelegate.java

```diff
@@ -117,10 +117,17 @@ private static void validateCloudSdk() throws CoreException  {
   @Override
   public ILaunch getLaunch(ILaunchConfiguration configuration, String mode) throws CoreException {
     IServer thisServer = ServerUtil.getServer(configuration);
-    DefaultRunConfiguration thisConfig = generateServerRunConfiguration(configuration, thisServer);
+    return getLaunchInternal(configuration, mode, thisServer, getLaunchManager());
```

Comment:
Could we make this less test-centric and instead something like:
```
IServer thisServer = ServerUtil.getServer(configuration);
checkConflictingLaunches(configuration, thisServer, launchManager.getLaunches()); 
return super.getLaunch(configuration, mode);
```
Could even pass in the `DefaultRunConfiguration` instance instead and avoid passing in the `IServer`.